sources/Inputs/include/Contingencies.h

```diff
@@ -0,0 +1,137 @@
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace dfl {
+namespace inputs {
+
+/**
+ * @brief Contingency elements
+ */
+struct ContingencyElement {
+  /**
+   * @brief Enum with accepted types for the elements in a contingency
+   */
+  enum class Type {
+    LOAD = 0,
+    GENERATOR,
+    BRANCH,
+    LINE,
+    TWO_WINDINGS_TRANSFORMER,
+    THREE_WINDINGS_TRANSFORMER,
+    SHUNT_COMPENSATOR,
+    STATIC_VAR_COMPENSATOR,
+    DANGLING_LINE,
+    HVDC_LINE,
+    BUSBAR_SECTION
+  };
+
+  /**
+   * @brief Constructor
+   *
+   * @param id the element id
+   * @param type the element type
+   */
+  ContingencyElement(const std::string& id, Type type) : id(id), type(type) {}
+
+  /**
+   * @brief Check if a given type is valid against a reference type
+   *
+   * Used to check element types observed in the network against types used in input contingencies
+   *
+   * @param type a type to check
+   * @param referenceType the reference type to check against
+   */
+  static bool isValidType(Type type, Type referenceType);
+
+  /**
+   * @brief Get type enum value from a string
+   *
+   * Parses a string into its enum value
+   *
+   * @return none if not a valid type, otherwise the enum value
+   */
+  static boost::optional<Type> typeFromString(const std::string& str);
+
+  /**
+   * @brief Type to string
+   *
+   * Transforms a type enum value into its string representation
+   *
+   * @return string representation of element type
+   */
+  static std::string toString(Type type);
+
+  std::string id;  ///< Identifier of an element affected by a contingency
+  Type type;       ///< Type of the element affected by the contingency (BRANCH, GENERATOR, LOAD, ...)
+};
+
+/**
+ * @brief Contingency
+ */
+struct Contingency {
+  /**
+   * @brief Constructor
+   *
+   * @param id the contingency id
+   */
+  explicit Contingency(const std::string& id) : id(id), elements{} {}
+
+  std::string id;                            ///< Identifier of the contingency
+  std::vector<ContingencyElement> elements;  ///< Elements affected by the contingency
+};
+
+/**
+ * @brief Manage the contingencies given as input for a Security Analysis simulation
+ */
+class ContingenciesManager {
+ public:
+  /**
+   * @brief Constructor
+   *
+   * Load contingency from file. Exit the program on error in parsing the file
+   *
+   * @param filepath the JSON contingencies file to use
+   */
+  explicit ContingenciesManager(const boost::filesystem::path& filepath);
+
+  /**
+   * @brief List of contingencies
+   *
+   * Obtain the list of contingencies defined in the input
+   *
+   * @return contingency list
+   */
+  std::shared_ptr<const std::vector<Contingency>> get() const {
+    return contingencies_;
+  }
+
+ private:
+  /// @brief Load contingencies from an input file
+  void load(const boost::filesystem::path& filepath);
+
+  std::shared_ptr<std::vector<Contingency>> contingencies_;  ///< Contingencies obtained from input file
```

Comment:
The vector is at worst empty but not "not existing" so using a pointer here is confusing